suorituspalvelu-ui/src/api.ts

```diff
@@ -0,0 +1,61 @@
+import { isEmpty, isNullish, omitBy } from 'remeda';
+import { configPromise } from './configuration';
+import { client } from './http-client';
+
+export type SearchParams = {
+  oppija?: string;
+  oppilaitos?: string;
+  vuosi?: string;
+  luokka?: string;
+};
+
+type SearchResult = {
+  oppijat: Array<{
+    oppijaNumero: string;
+    hetu?: string;
+    nimi: string;
+  }>;
+};
+
+export const cleanSearchParams = (params: SearchParams) => {
+  return omitBy(params, (value) => isEmpty(value) || value === '');
+};
+
+export const searchOppijat = async (params: SearchParams) => {
```

Comment:
On ehkä vähän hassua et kaikki APIin liittyvä on samassa tiedostossa mut ilmeisesti tän APIn käyttö on aika yksinkertaista niin ehkäpä se on perusteltua.

Reply:
Juuri tuo ajatus oli itsellä, että en lähtisi tässä vaiheessa jakamaan näitä erillisiin tiedostoihin, kun näitä on niin vähän.